wren-ui/src/hooks/useNativeSQL.tsx

```diff
@@ -0,0 +1,48 @@
+import { useState } from 'react';
+import { useGetSettingsQuery } from '@/apollo/client/graphql/settings.generated';
+import { useGetNativeSqlLazyQuery } from '@/apollo/client/graphql/home.generated';
+import { DataSourceName } from '@/apollo/client/graphql/__types__';
+
+export interface NativeSQLResult {
+  data: string;
+  dataSourceType: DataSourceName;
+  hasNativeSQL: boolean;
+  loading: boolean;
+}
+
+// we assume that not having a sample dataset means supporting native SQL
+function useNativeSQLInfo() {
+  const { data: settingsQueryResult } = useGetSettingsQuery();
```

Comment:
We can use this API for now. But if the settings API designs more properties in the future, I think we will need to provide another API ? 🤔

Reply:
Yep, I think so.